### PR TITLE
Test cases for XIS - serializer, views and setup

### PR DESCRIPTION
**Pull request by KarenAJ**
@OpenLXP/contributors Please do review the test cases added in the XIS implementation.

### CODE REVIEW

File: app/api/urls.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-from api import views
-
-router = DefaultRouter()
-
-urlpatterns = [
-    path('metadata-ledger/', views.MetadataLedgerView.as_view()),
-    path('metadata/', views.MetadataLedgerView.as_view()),
-]
\ No newline at end of file
+from django.urls import path
```
Comment (mattdrayer) [inline app/api/urls.py:1]: These file changes aren't really clear to me -- it looks like we are essentially replacing the code with the same code -- any idea what is going on?

File: app/api/tests/test_setup.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+from rest_framework.test import APITestCase
+from django.urls import reverse
+from core.models import XISConfiguration
+
+
+class TestSetUp(APITestCase):
+    """Class with setup and teardown for tests in XIS"""
+
+    def setUp(self):
```
Comment (jdiouf) [inline app/api/tests/test_setup.py:9]: I would make sure the data here is all random/contains no data from an actual record. Also, maybe we can look into a way of creating the data dynamically (don't have to address this now)

File: docker-compose.yml [hunk 3]
```diff
@@ -41,23 +41,7 @@ services:
...
       - openlxp
-  es01:
-    image: docker.elastic.co/elasticsearch/elasticsearch:7.11.1
-    container_name: es01
-    environment:
-      - discovery.type=single-node
-    ulimits:
-      memlock:
-        soft: -1
-        hard: -1
```
Comment (jdiouf) [inline docker-compose.yml:52]: I think your push overwrote Connor's elastic search configuration. We need to undo this before we can merge.

File: app/api/tests/test_setup.py (hunk 2 above)
Comment (mattdrayer) [reply app/api/tests/test_setup.py:9]: Ref: https://ddt.readthedocs.io/en/latest/example.html

File: docker-compose.yml (hunk 3 above)
Comment (KarenAJ) [reply docker-compose.yml]: Good catch, I think I may have messed with the code during rebasing. I tried fixing it, please do check and see

File: app/api/urls.py (hunk 1 above)
Comment (KarenAJ) [reply app/api/urls.py]: No sure either Matt, I think I messed around with edits during rebasing and could be that.

File: app/api/tests/test_setup.py (hunk 2 above)
Comment (KarenAJ) [reply app/api/tests/test_setup.py:9]: Sure @jdiouf . Thanks for this @mattdrayer

File: docker-compose.yml [hunk 4]
```diff
@@ -1,18 +1,20 @@
 version: "3"
 
 services:
-  db:
+  db_xis:
     image: mysql:5.7
     ports:
-      - '3306:3306'
+      - '3310:3306'
```
Comment (jdiouf) [inline docker-compose.yml:7]: Was there a conflict with the old port?

File: docker-compose.yml [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
 #      MYSQL_USER: 'root'
       MYSQL_PASSWORD: "${DB_PASSWORD}"
       MYSQL_ROOT_PASSWORD: "${DB_ROOT_PASSWORD}"
       MYSQL_HOST: ''
-
-  app:
+    networks:
+      - openlxp
+  app_xis:
+    container_name: openlxp-xis-load
```
Comment (jdiouf) [inline docker-compose.yml:17]: I would just name the container openlxp-xis unless it causes an issue.

**Review by jdiouf — APPROVED** [review]

**Review by mattdrayer — APPROVED** [review]
Aside from the oddly-changed files which appear to not have any actual changes to them, the PR looks fine.  It would be good to understand what has happened with this branch -- perhaps an interactive rebase that includes squashing of the commits will resolve the issue.

**Pull request merged**